### PR TITLE
New literal_coercion_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #962

### CODE REVIEW

**Comment by MichaelChirico** [comment]
`tibble` is dotted-line Suggested because `patrick` imports it, so I'll add it to our Suggests directly given this:

>  '::' or ':::' import not declared from: ‘tibble’

File: DESCRIPTION [hunk 1]
```diff
@@ -26,6 +26,7 @@ Imports:
     jsonlite,
     knitr,
     stats,
+    tibble,
```
Comment (AshesITR) [inline DESCRIPTION:29]: Why import?

Comment (MichaelChirico) [reply DESCRIPTION]: whoops

File: inst/lintr/linters.csv [hunk 2]
```diff
@@ -26,6 +26,7 @@ if_else_match_braces_linter,default style readability
 implicit_integer_linter,style consistency best_practices
 infix_spaces_linter,style readability default
 line_length_linter,style readability default configurable
+literal_coercion_linter,best_practices efficiency
```
Comment (AshesITR) [inline inst/lintr/linters.csv:29]: Maybe also consistency?

File: R/literal_coercion_linter.R [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+#'   necessary to specify the storage type of `NA`, but when it is, prefer
+#'   using the typed version (e.g. `NA_real_`) instead of a coercion
+#'   (like `as.numeric(NA)`).
+#'
+#' @evalRd rd_tags("literal_coercion_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+literal_coercion_linter <- function() {
+  Linter(function(source_file) {
+    if (length(source_file$parsed_content) == 0L) {
```
Comment (AshesITR) [inline R/literal_coercion_linter.R:16]: require xml

**Review by AshesITR — APPROVED** [review]

**Pull request merged**